install/kubernetes/cilium/templates/cilium-operator/clusterrole.yaml

```diff
@@ -7,26 +7,43 @@ rules:
 - apiGroups:
   - ""
   resources:
-  # to automatically delete [core|kube]dns pods so that are starting to being
-  # managed by Cilium
   - pods
   verbs:
   - get
   - list
   - watch
+{{- if hasKey .Values "disableEndpointCRD" }}
+{{- if (eq (.Values.disableEndpointCRD | quote ) ( "false" | quote )) }}
```

Comment:
I don't think the logic is quite right here.

if (endpoint CRDs are enabled) {
 ... add delete permission ...
} else if (restart unmanaged pods) {
... add delete permission ...
}

This means that you have to both disable endpoint CRD _and_ disable unmanaged pods restart. But if endpoint CRDs are enabled but restarting pods is disabled, the permission will still be there unnecessarily.

Reply:
Discussed offline, fixed in the PR.